### DISCUSSION TITLE
How to get the force applied by the distance constraint?

### DISCUSSION THREAD

**Discussion by Ewall198** [jrouwe/JoltPhysics#1301] [Q&A] [answered]
I have a distance constraint between a `body_a: StaticBody` and a `body_b: DynamicBody` (moveable). How do I get the total force applied to `body_b` by the constraint.

I tried using `GetTotalLambdaPosition`, but this gives unexpected results. When `body_b` is dangling in the air, `GetTotalLambdaPosition` returns a value of zero *sometimes*. I'd expect the value to be non-zero since it is applying enough force to hold `body_b` in the air. Any ideas?

**Comment by jrouwe** [comment, marked as answer]
`GetTotalLambdaPosition` is the correct call. It could be that you have an oscillation in the system. If the current distance of the distance constraint is between min and max distance then no force will be applied until either the min or max distance is violated again.

**Reply by Ewall198** [reply]
Ok thanks I was getting inconsistent behavior due to a bug in my code. But, now that I fixed it, `GetTotalLambdaPosition / time_step` is providing the expected force. Thank you!
